src/types/state.js

```diff
@@ -34,9 +34,13 @@ import type { TransformStacksPerThread } from './transforms';
 import type JSZip from 'jszip';
 import type { IndexIntoZipFileTable } from '../profile-logic/zip-files';
 import type { PathSet } from '../utils/path.js';
+import type { ProfileData as ImportedProfileData } from 'firefox-profiler/app-logic/published-profiles-store';
 
 export type Reducer<T> = (T | void, Action) => T;
 
+// Reexport this type for easier access
+export type ProfileData = ImportedProfileData;
```

Comment:
I think that now this type is used more widely, it would be better to support it properly:


* I would prefer the type definition moved to `src/types`, rather than re-exported.
* I think this type could use a better name. It's a bit ambiguous, especially grepping across the project. (See `import * as ProfileData from '../../profile-logic/profile-data';`)
* The type has no documentation explaining what it is.

I bring all this up, as I had to do a bit of digging to really figure out what was going on here, and what this ProfileData was.

Reply:
I'll add more comment and, yeah, it's a good idea to rename the type But I think the type definition should stay where it is now.
Indeed, the data in IndexedDB will persist across reloads, and therefore it's important that its typing is controlled strictly. On the other hand the data in the reducer is valid only during one run. So the concern I have is that if we define this type in the state there's a high probability we'll change it without doing any IDB check and upgrade.